src/main/java/com/mercedesbenz/sechub/sdk/settings/AppSettingsConfigurable.java

```diff
@@ -0,0 +1,107 @@
+package com.mercedesbenz.sechub.sdk.settings;
+
+import com.intellij.credentialStore.Credentials;
+import com.intellij.openapi.options.Configurable;
+import com.mercedesbenz.sechub.plugin.idea.sechubaccess.SecHubAccess;
+import com.mercedesbenz.sechub.plugin.idea.sechubaccess.SecHubAccessSupport;
+import com.mercedesbenz.sechub.plugin.idea.window.SecHubServerPanel;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.util.Objects;
+
+/*
+ * Provides controller functionality for application settings.
+ */
+final class AppSettingsConfigurable implements Configurable {
+
+    private AppSettingsComponent appSettingsComponent;
+    private AppSettingsCredentialsSupport appSettingsCredentialsSupport;
+
+    // A default constructor with no arguments is required because
+    // this implementation is registered as an applicationConfigurable
+
+    @Nls(capitalization = Nls.Capitalization.Title)
+    @Override
+    public String getDisplayName() {
+        return "SecHub";
+    }
+
+    @Override
+    public JComponent getPreferredFocusedComponent() {
+        return appSettingsComponent.getPreferredFocusedComponent();
+    }
+
+    @Nullable
+    @Override
+    public JComponent createComponent() {
+        appSettingsComponent = new AppSettingsComponent();
+        appSettingsCredentialsSupport = new AppSettingsCredentialsSupport();
+        return appSettingsComponent.getPanel();
+    }
+
+    @Override
+    public boolean isModified() {
+        AppSettings.State state = Objects.requireNonNull(AppSettings.getInstance().getState());
+        String currentPassword = "";
+        String userName = "";
+
+        Credentials credentials = appSettingsCredentialsSupport.retrieveCredentials();
+        if (credentials != null) {
+            currentPassword = credentials.getPasswordAsString();
+            userName = credentials.getUserName();
+        }
+
+        return !appSettingsComponent.getUserNameText().equals(userName) ||
+                appSettingsComponent.getApiTokenPassword().equals(currentPassword) ||
+                appSettingsComponent.getServerUrlText().equals(state.serverURL);
+    }
+
+    @Override
+    public void apply() {
+        AppSettings.State state = Objects.requireNonNull(AppSettings.getInstance().getState());
+        state.serverURL = appSettingsComponent.getServerUrlText();
+
+        Credentials credentials = new Credentials(appSettingsComponent.getUserNameText(), appSettingsComponent.getApiTokenPassword());
+        appSettingsCredentialsSupport.storeCredentials(credentials);
+
+        updateComponents(state);
+    }
+
+
+
+    @Override
+    public void reset() {
```

Comment:
This reset mechanism sets the values to the previously configured values. Wouldn't it be better rest would clear the entries?

Update: I'm not sure if it is a good idea, if we allow like empty strings as a values for the reset button. I don't know if this triggers the same validation as the apply method. Then we'd have to allow empty strings if someone wants to clear they data. You can decide whats the best, I just thought it might be a good idea to provide the possibility to clear the data.

Reply:
can not do full reset in reset method as it is called when initialize the IDE -> credentials would never be stored and have to be configured everytime starting the IDE :-1: (reset is from the interface)